app/uk/gov/hmrc/serviceconfigs/persistence/DeploymentConfigSnapshotRepository.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.serviceconfigs.persistence
+
+import com.mongodb.BasicDBObject
+import org.mongodb.scala.bson.{BsonDocument, BsonInt32}
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.PlayMongoRepository
+import uk.gov.hmrc.serviceconfigs.model.DeploymentConfigSnapshot
+import org.mongodb.scala.model.Filters.{and, equal}
+import org.mongodb.scala.model.FindOneAndReplaceOptions
+import org.mongodb.scala.model.Indexes._
+import org.mongodb.scala.model.{IndexModel, IndexOptions}
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class DeploymentConfigSnapshotRepository @Inject()(mongoComponent: MongoComponent)(implicit ec: ExecutionContext)
+  extends PlayMongoRepository(
+    mongoComponent = mongoComponent,
+    collectionName = "deploymentConfigSnapshots",
+    domainFormat = DeploymentConfigSnapshot.mongoFormat,
+    indexes = Seq(
+      IndexModel(hashed("name"), IndexOptions().background(true)),
+      IndexModel(hashed("environment"), IndexOptions().background(true)))
+  ) {
+
+  def snapshotsForService(name: String): Future[Seq[DeploymentConfigSnapshot]] = {
+    val sortParams =
+      BsonDocument(
+        ("date", BsonInt32(1))
+      )
+
+    collection
+      .find(equal("name", name))
+      .sort(sortParams)
```

Comment:
maybe just inline the sort params? and can use the helpers to clarify what `1` means  `.sort(Sorts.ascending("date"))`

Reply:
Oh wow, that's much better.